src/components/common/CodeReview.js

```diff
@@ -485,23 +527,35 @@ class CodeReview extends Component {
             lineNumber={i + 1}
             onClickAdd={(lineNum) => this.handleAddComment(lineNum)}
             disableButton={isFeedbackView}
+            numThreads={numThreads}
+            toggleHiddenStateCallback={this.toggleHiddenCommentState}
           />
-          {renderedComments}
+          {hiddenCommentState ? null : renderedComments}
         </Fragment>
       );
     });
   }
 
   render() {
-    const { classes, code } = this.props;
-    const { comments } = this.state;
-
+    const { classes, code, isStudentView, isFeedbackView } = this.props;
+    const { comments, lineCommentsHiddenState } = this.state;
+    // consider also bot and teacher comment !
```

Comment:
Is this a `todo`?

Reply:
Yes actually the button should appear only if there are comments in the review. But at the moment i only count the length of `comments` which is just student comments. So i have to also count bot- and teacher-comments